sdk/search/Azure.Search/src/Utilities/GeneratorTweaks.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using Azure.Search.Models;
+
+#pragma warning disable SA1402 // File may only contain a single type
+
+namespace Azure.Search
+{
+    // Hide the unused ServiceClient class.
+    [CodeGenClient("")]
+    internal partial class ServiceClient { }
+
+    // Hide the unused DocumentsClient class.
+    [CodeGenClient("Documents")]
+    internal partial class DocumentsClient { }
+
+    // Work-around the generator not enjoying mixing model types between the
```

Comment:
https://github.com/Azure/autorest.csharp/issues/486

Reply:
Awesome - added a comment to help track